brainbuilder/utils/sonata/split_population.py

```diff
@@ -661,8 +661,9 @@ def _write_subcircuit_external(output, circuit, id_mapping):
             if len(wanted_src_ids) == 0:
                 continue
 
-            new_name = f"external_{name}"
-            output_path = output / (new_name + ".h5")
+            new_node_name = f"external_{edge.source.name}_{edge.target.name}"
```

Comment:
Hmm, an external node population name should not depend on `edge.target.name` I believe. E.g., there should only be one external population `external_A` which may have corresponding edge populations `external_A__A`,  `external_A__B`, `external_A__C`, etc.
```suggestion
            new_node_name = f"external_{edge.source.name}"
```

Reply:
Also, it should be assured that `external_A` does not already exist (e.g., from some earlier extraction), in which case it should be renamed to `external_external_A` etc. (Michael had some code for that.)

Reply:
> there should only be one external population external_A

That is a choice that has to be made - by having multiple `external_A_*`, then it makes it easier to enable or not enable them (or include them in the circuit config) - this all depends on whether a particular set of edge intersect with the the source set: https://github.com/OpenBrainInstitute/brainbuilder/blob/f96500cbf04cf32c8718995efa41e73bb8425eea/brainbuilder/utils/sonata/split_population.py#L641

If there is only one `external_A`, then there may be source ids that have no effect on edge population `A__B` because they only impact `A__C`.

>  Michael had some code for that.

That would happen in a different commit; this is fixing the attribute not being correctly created.

Reply:
> That is a choice that has to be made - by having multiple external_A_*, then it makes it easier to enable or not enable them (or include them in the circuit config)

I would keep it as Michael had it in his implementation. Which is having only one `external_A` population if I am not mistaken. Otherwise, things are getting really complicated (and unintuitive) if population A would be further subdivided into smaller pieces IMO.

> If there is only one external_A, then there may be source ids that have no effect on edge population A__B because they only impact A__C.

Hmm, not sure I understand why this would be undesirable. Maybe I am missing something. I think this is perfectly fine if not all source IDs from A are actually creating any connections to B.

> That would happen in a different commit; this is fixing the attribute not being correctly created.

Ok, I see.